src/libmeasurement_kit/common/logger.cpp

```diff
@@ -210,41 +209,25 @@ class DefaultLogger : public Logger, public NonCopyable, public NonMovable {
 
     void emit_event_ex(
             const std::string &key, nlohmann::json &&value) override {
-        if (!value.is_object()) {
-            warn("wrong value for key: %s", key.c_str());
-            assert(false);
-        }
-#ifndef NDEBUG
-        {
-            bool found = false;
-            do {
-#define CHECK(name_, type_ignored_, mandatory_ignored_)                        \
-    if (key == #name_) {                                                       \
-        found = true;                                                          \
-        break;                                                                 \
-    }
-                MK_ENUM_EVENTS(CHECK)
-#undef CHECK
-            } while (0);
-            if (!found) {
-                fprintf(stderr, "PANIC: unknown event: %s\n", key.c_str());
-                abort();
-            }
-        }
-#endif
-        if (handlers_.count(key) <= 0) {
-            return;
-        }
         nlohmann::json event{
             {"key", key},
             {"value", std::move(value)}
         };
+        std::string real_key;
+        if (handlers_.count(key) <= 0) {
+            // Even if the event has not registered handler, we pass them up
+            // one layer to validate the event structure.
+            real_key = "__disabled";
+            assert(handlers_.count(key));
```

Comment:
This `assert()` is here to make sure I have actually registered a handler for unhandled messages.